EIPS/eip-6617.md

```diff
@@ -14,12 +14,11 @@ created: 2023-02-27
 
 This EIP offers a standard for building a bit-based permission and role system. Each permission is represented by a single bit. By using an `uint256`, up to $256$ permissions and $2^{256}$ roles can be defined. We are able to specify the importance of each permission based on the order of the bits.
 
-
 ## Specification
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
 
-*Note* The following specifications use syntax from Solidity `0.8.7` (or above)
+_Note_ The following specifications use syntax from Solidity `0.8.7` (or above)
 
```

Comment:
So it looks like this standard is just a bitfield? Like there's no interface to read or interact with the `uint256`, so there's no way to check if a particular user has or doesn't have a permission?

Because of that, I think this proposal is too incomplete to move into the Review status.

Reply:
@SamWilsn We tried to separate this proposal from #6366 following the suggestion from @Pandapip1. This proposal is describing how could we use the `uint256` to store permissions and roles instead of `string` or `bytes32`.

> So it looks like this standard is just a bitfield? Like there's no interface to read or interact with the `uint256`, so there's no way to check if a particular user has or doesn't have a permission?

We implemented this EIP as a library, with several methods to manipulate a subset of bits which were stored in `uint256`

https://github.com/ethereum/EIPs/blob/301d3223bd0aa38836bb73509adaae949db5c23e/EIPS/eip-6617.md?plain=1#L107-L113

This method is used to identify a certain bits in `_permission` is a superset of bits in `_requiredPermission` by which we can tell the user has the permissions or not.

cc @vdusart

Reply:
Ok thanks for the feedbacks @SamWilsn, is it better like that ?